[18.0][MIG] partner_label: Migration to 18.0

Kimkhoi3010 opened the pull request:
Migrate `partner_label` to 18.0
## Depend on:
- `https://github.com/OCA/partner-contact/pull/1696`

hbrunn committed:
[ADD] partner_label

hbrunn committed:
fixup! [ADD] partner_label

benwillig committed:
[FIX] Fixed labels alignment and renamed report to 'Partner Labels'

benwillig committed:
[IMP] Allow to specify margin for each side (top, bottom, ...)

hbrunn committed:
[MIG][12.0] partner_label

ajayjaviya committed:
[MIG][13.0] partner_label

MiquelRForgeFlow committed:
[ADD] purchase_supplier_rank

alexeirivera87 committed:
[IMP] partner_label: black, isort, prettier

alexeirivera87 committed:
[MIG] partner_label: Migration to 14.0

emagdalenaC2i committed:
[MIG] partner_label: Migration to 15.0

emagdalenaC2i committed:
[FIX] fix view

Quan-nhm committed:
[15.0][MIG] partner_label: migrate to 15.0

amarcosg committed:
[MIG] purchase_supplier_rank: Migration to 16.0

florian-dacosta committed:
[IMP] Rename partner_helper to partner_address_split + add some tests

AlexPForgeFlow committed:
[MIG] partner_label: Migration to 16.0

JordiMForgeFlow committed:
[ADD] partner_identification_eori

Commit:
[UPD] Update partner_label.pot

OCA-git-bot committed:
[BOT] post-merge updates

Ivorra78 committed:
Translated using Weblate (Spanish)

Currently translated at 100.0% (21 of 21 strings)

Translation: partner-contact-16.0/partner-contact-16.0-partner_label
Translate-URL: https://translation.odoo-community.org/projects/partner-contact-16-0/partner-contact-16-0-partner_label/es/

EmilioPascual committed:
[ADD] partner_purchase_manager

Add purchase manager in partner model.

MT-4252 @moduon

Commit:
[UPD] Update partner_purchase_manager.pot

OCA-git-bot committed:
[BOT] post-merge updates

mymage committed:
Added translation using Weblate (Italian)

mymage committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (2 of 2 strings)

Translation: partner-contact-16.0/partner-contact-16.0-partner_identification_eori
Translate-URL: https://translation.odoo-community.org/projects/partner-contact-16-0/partner-contact-16-0-partner_identification_eori/it/

mymage committed:
Added translation using Weblate (Italian)

mymage committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (1 of 1 strings)

Translation: partner-contact-16.0/partner-contact-16.0-purchase_supplier_rank
Translate-URL: https://translation.odoo-community.org/projects/partner-contact-16-0/partner-contact-16-0-purchase_supplier_rank/it/

mymage committed:
Translated using Weblate (Italian)

Currently translated at 28.5% (6 of 21 strings)

Translation: partner-contact-16.0/partner-contact-16.0-partner_label
Translate-URL: https://translation.odoo-community.org/projects/partner-contact-16-0/partner-contact-16-0-partner_label/it/

mymage committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (21 of 21 strings)

Translation: partner-contact-16.0/partner-contact-16.0-partner_label
Translate-URL: https://translation.odoo-community.org/projects/partner-contact-16-0/partner-contact-16-0-partner_label/it/

mymage committed:
Added translation using Weblate (Italian)

mymage committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (3 of 3 strings)

Translation: partner-contact-16.0/partner-contact-16.0-partner_purchase_manager
Translate-URL: https://translation.odoo-community.org/projects/partner-contact-16-0/partner-contact-16-0-partner_purchase_manager/it/

WorrapongOngsakul committed:
[IMP] partner_label: pre-commit auto fixes

WorrapongOngsakul committed:
[MIG] partner_label: Migration to 17.0

MarinaAForgeFlow committed:
[ADD] partner_shipping_policy

yankinmax committed:
[17.0][MIG] partner_address_split: Migration to 17.0

mymage committed:
Added translation using Weblate (Italian)

mymage committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (1 of 1 strings)

Translation: partner-contact-17.0/partner-contact-17.0-partner_address_split
Translate-URL: https://translation.odoo-community.org/projects/partner-contact-17-0/partner-contact-17-0-partner_address_split/it/

Commit:
[IMP] partner_contact_role: pre-commit execution

Commit:
[UPD] Update partner_shipping_policy.pot

OCA-git-bot committed:
[BOT] post-merge updates

mymage committed:
Added translation using Weblate (Italian)

mymage committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (5 of 5 strings)

Translation: partner-contact-16.0/partner-contact-16.0-partner_shipping_policy
Translate-URL: https://translation.odoo-community.org/projects/partner-contact-16-0/partner-contact-16-0-partner_shipping_policy/it/

LucasTran380381 committed:
[MIG] partner_address_street3: Migration to 18.0

LucasTran380381 committed:
[IMP] partner_address_street3: pre-commit auto fixes

Commit:
[MIG] partner_address_street3: Migration to 18.0

CRogos committed:
[ADD] partner_type_base: improve partner type customizable

CRogos committed:
[IMP] partner_firstname: adapt to partner_type_base changes

BhaveshHeliconia committed:
[IMP] partner_address_split: pre-commit auto fixes

BhaveshHeliconia committed:
[MIG] partner_address_split: Migration to 18.0

BhaveshHeliconia committed:
[IMP] partner_identification_eori: pre-commit auto fixes

BhaveshHeliconia committed:
[MIG] partner_identification_eori: Migration to 18.0

BhaveshHeliconia committed:
[IMP] purchase_supplier_rank: pre-commit auto fixes

BhaveshHeliconia committed:
[MIG] purchase_supplier_rank: Migration to 18.0

BhaveshHeliconia committed:
[MIG] partner_contact_nationality: Migration to 18.0

BhaveshHeliconia committed:
[IMP] partner_purchase_manager: pre-commit auto fixes

BhaveshHeliconia committed:
[MIG] partner_purchase_manager: Migration to 18.0

BhaveshHeliconia committed:
[IMP] partner_shipping_policy: pre-commit auto fixes

BhaveshHeliconia committed:
[MIG] partner_shipping_policy: Migration to 18.0

sergio-teruel committed:
[IMP] base_location: Add city code to be used for other purposes

apami0111 committed:
[IMP]partner_industry_secondary - test_res_partner_industry.py
- Replaced assertRaises with assertRaisesRegex to validate specific error messages.
- Added docstrings to each test case for better clarity and documentation.

MarinaAForgeFlow committed:
[ADD] partner_supplier_ref_sequence

Kimkhoi3010 committed:
[IMP] partner_label: pre-commit auto fixes

Kimkhoi3010 committed:
[MIG] partner_label: Migration to 18.0

nguyenminhchien reviewed (changes requested):
run pre-commit command

xaviedoanhduy commented on partner_label/views/base_config_settings.xml:
```diff
@@ -0,0 +1,116 @@
+        <field name="model">res.config.settings</field>
+        <field name="inherit_id" ref="base_setup.res_config_settings_view_form" />
+        <field name="arch" type="xml">
+            <xpath expr="//div[@id='companies']" position="after">
+                <div id="partner_labels">
+                    <h2>Labels configuration</h2>
+                    <div class="row mt16 o_settings_container">
+                        <div class="col-12 col-lg-6 o_setting_box">
+                            <div class="o_setting_right_pane">
+                                <div class="content-group">
+                                    <div class="mt16 row">
```
```suggestion
            <xpath expr="//div[@id='companies']" position="after">
                <div id="partner_labels">
                    <h2>Labels configuration</h2>
                    <div class="row mt16 o_settings_container">
                        <div class="col-12 col-lg-6 o_setting_box">
                            <div class="o_setting_right_pane">
                                <div class="content-group">
                                    <div class="mt16 row">
```

> ```
> Settings views (those inheriting from res.config.settings) have been simplified. Check for the details at https://github.com/odoo/odoo/pull/106425.
> 
> Coming from previous versions, the needed changes are:
> 
> The <div class="app_settings_block *" ...> element is now <app ...>.
> The <div class="o_settings_container *" ...> is now <block ...>.
> 
> The <h2> title is now filled in the title attribute inside the <block> tag.
> The <div class="o_setting_box *" ...> is now <setting ...>.
> The divs with classes o_setting_right_pane or o_setting_left_pane are no more needed and should be removed.
> The fields, labels, paragraphs are put inside <setting> tag.
> ```
> 

see: https://github.com/OCA/maintainer-tools/wiki/Migration-to-version-17.0

xaviedoanhduy commented on partner_label/views/base_config_settings.xml:
```diff
@@ -0,0 +1,116 @@
+                    <block title="Labels configuration" />
+                    <div class="row mt16 o_settings_container">
+                        <div class="col-12 col-lg-6 o_setting_box">
+                            <div class="o_setting_right_pane">
```
```suggestion
                    <div class="row mt16 o_settings_container">
                        <div class="col-12 col-lg-6 o_setting_box">
                            <div class="o_setting_right_pane">
```
```
> The <div class="o_setting_box *" ...> is now <setting ...>.
> The divs with classes o_setting_right_pane or o_setting_left_pane are no more needed and should be removed.
> The fields, labels, paragraphs are put inside <setting> tag.
```
Let's look at an example from the commit I mentioned: `https://github.com/odoo/odoo/pull/106425`

xaviedoanhduy reviewed (changes requested):
btw please remove changes to modules not related to this migration

xaviedoanhduy commented on partner_label/views/base_config_settings.xml:
```diff
@@ -0,0 +1,98 @@
+            <xpath expr="//div[@id='companies']" position="after">
+                <div id="partner_labels">
+                    <block
+                        title="Labels configuration"
```
```suggestion
                        title="Labels Configuration"
```

xaviedoanhduy reviewed (changes requested):
looks weird :slightly_smiling_face:

<img width="1083" height="582" alt="image" src="https://github.com/user-attachments/assets/a1639817-3ff7-4378-a5cc-3b030d04f687" />


fyi, this is the layout in version 16.0:

<img width="1041" height="408" alt="image" src="https://github.com/user-attachments/assets/e10506fa-f800-48aa-81ba-700cac2a5d6f" />

xaviedoanhduy reviewed (changes requested):
maybe because the font in each browser is different, leading to different layout

xaviedoanhduy commented on partner_label/views/base_config_settings.xml:
```diff
@@ -0,0 +1,126 @@
+                                        for="partner_labels_paperformat_id"
+                                        class="col-lg-3 o_light_label"
+                                    />
+                                    <field name="partner_labels_paperformat_id" />
```
```suggestion
                                    <field name="partner_labels_paperformat_id" class="oe_inline" />
```

xaviedoanhduy commented on partner_label/views/base_config_settings.xml:
```diff
@@ -0,0 +1,126 @@
+                                    type="object"
+                                    name="action_partner_labels_preview"
+                                    string="Preview"
+                                    class="oe_link"
```
```suggestion
                                    class="oe_link"
                                    icon="oi-arrow-right"
```
same with buttons in config setting

xaviedoanhduy reviewed (approved):
LG
pre-approve

legalsylvain committed:
[IMP] partnerfirstname: introduce firstname.mixin
Mutualize common things used by models that have firstname / lastname fields.

legalsylvain committed:
[IMP] partner_firstname: add option to make fields required
* add option 'partner_names_required_fields' at res.config.settings level.
* adapt partner and user views, displaying required field regarding configuration.
* improve configure documentation.

yankinmax committed:
[18.0][ADD] partner_rank_single

OCA-git-bot committed:
Merge PR #2182 into 18.0

Signed-off-by pedrobaeza

Commit:
[UPD] Update base_location.pot

OCA-git-bot committed:
[BOT] post-merge updates

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: partner-contact-18.0/partner-contact-18.0-base_location
Translate-URL: https://translation.odoo-community.org/projects/partner-contact-18-0/partner-contact-18-0-base_location/

OCA-git-bot committed:
Merge PR #2170 into 18.0

Signed-off-by yajo

OCA-git-bot committed:
[BOT] post-merge updates

mymage committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (35 of 35 strings)

Translation: partner-contact-18.0/partner-contact-18.0-base_location
Translate-URL: https://translation.odoo-community.org/projects/partner-contact-18-0/partner-contact-18-0-base_location/it/

AungKoKoLin1997 committed:
[IMP] partner_contact_address_default: add option to allow selecting any partner

ivantodorovich committed:
[FIX] partner_property: broken partner/user export

Steps to reproduce:

1. Go in the partner or users list view
2. Select a few records
3. Click on Actions > Export

Expected:

The Export wizard should open.

Result:

```
File "/odoo/src/odoo/addons/web/controllers/export.py", line 400, in get_fields
    exportable_fields.update(self._get_property_fields(fields, model, domain=domain))
                            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
File "/odoo/src/odoo/addons/web/controllers/export.py", line 321, in _get_property_fields
    field_to_get = Model._field_to_sql(Model._table, definition_record, self_subquery)
                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
File "/odoo/src/odoo/odoo/models.py", line 2976, in _field_to_sql
    raise ValueError(f"Cannot convert {field} to SQL because it is not stored")
ValueError: Cannot convert res.partner.properties_company_id to SQL because it is not stored
```

OCA-git-bot committed:
Merge PR #2190 into 18.0

Signed-off-by pedrobaeza

OCA-git-bot committed:
[BOT] post-merge updates

OCA-git-bot committed:
Merge PR #2067 into 18.0

Signed-off-by pedrobaeza

OCA-git-bot committed:
[BOT] post-merge updates

OCA-git-bot committed:
Merge PR #2081 into 18.0

Signed-off-by hbrunn

Commit:
[UPD] Update partner_bank_code.pot

OCA-git-bot committed:
[BOT] post-merge updates

OCA-git-bot committed:
Merge PR #2181 into 18.0

Signed-off-by NL66278

Commit:
[UPD] Update partner_rank_single.pot

OCA-git-bot committed:
[BOT] post-merge updates

mymage committed:
Added translation using Weblate (Italian)

Shide committed:
[FIX] base_location_nuts: NUTS fields view in one line

With `base_vat` module installed, the fields appear disorganized and also shift the VAT field.

OCA-git-bot committed:
Merge PR #1883 into 18.0

Signed-off-by NL66278

Commit:
[UPD] Update partner_address_street3.pot

OCA-git-bot committed:
[BOT] post-merge updates

OCA-git-bot committed:
Merge PR #1988 into 18.0

Signed-off-by NL66278

Commit:
[UPD] Update partner_contact_type_end_user.pot

OCA-git-bot committed:
[BOT] post-merge updates

mymage committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (2 of 2 strings)

Translation: partner-contact-18.0/partner-contact-18.0-partner_rank_single
Translate-URL: https://translation.odoo-community.org/projects/partner-contact-18-0/partner-contact-18-0-partner_rank_single/it/

OCA-git-bot committed:
Merge PR #1993 into 18.0

Signed-off-by NL66278

Commit:
[UPD] Update animal.pot

OCA-git-bot committed:
[BOT] post-merge updates

OCA-git-bot committed:
Merge PR #2201 into 18.0

Signed-off-by rafaelbn

Commit:
[UPD] Update base_location_nuts.pot

OCA-git-bot committed:
[BOT] post-merge updates

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: partner-contact-18.0/partner-contact-18.0-base_location_nuts
Translate-URL: https://translation.odoo-community.org/projects/partner-contact-18-0/partner-contact-18-0-base_location_nuts/

ivantodorovich committed:
[FIX] partner_property: export of res.users

Originally attempted to fix it here: 36b93fa723f020e2d0f5e7e85166f3574982ce29

But that fix only worked for res.partners due to not properly using the alias
in the `_field_to_sql` method.

Then, when trying to export res.users (or any model that inherits from
res.partner), we'd get the following error:

```
  File ".../odoo/addons/web/controllers/export.py", line 324, in _get_property_fields
    definition_records = target_model.search_fetch(
                         ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File ".../odoo/odoo/models.py", line 1781, in search_fetch
    return self._fetch_query(query, fields_to_fetch)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File ".../odoo/odoo/models.py", line 4240, in _fetch_query
    rows = self.env.execute_query(query.select(*sql_terms))
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File ".../odoo/odoo/api.py", line 993, in execute_query
    self.cr.execute(query)
  File ".../odoo/odoo/sql_db.py", line 582, in execute
    return self._cursor.execute(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File ".../odoo/odoo/sql_db.py", line 357, in execute
    res = self._obj.execute(query, params)
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
psycopg2.errors.AmbiguousColumn: column reference "company_id" is ambiguous
LINE 1: ...TRUE) AND ("res_company"."id" IN (SELECT COALESCE(company_id...
```

OCA-git-bot committed:
Merge PR #2006 into 18.0

Signed-off-by NL66278

Commit:
[UPD] Update partner_shipping_policy.pot

OCA-git-bot committed:
[BOT] post-merge updates

OCA-git-bot committed:
Merge PR #1995 into 18.0

Signed-off-by NL66278

Commit:
[UPD] Update partner_contact_nationality.pot

OCA-git-bot committed:
[BOT] post-merge updates

OCA-git-bot committed:
Merge PR #2005 into 18.0

Signed-off-by NL66278

Commit:
[UPD] Update partner_purchase_manager.pot

OCA-git-bot committed:
[BOT] post-merge updates

OCA-git-bot committed:
Merge PR #1992 into 18.0

Signed-off-by NL66278

Commit:
[UPD] Update purchase_supplier_rank.pot

OCA-git-bot committed:
[BOT] post-merge updates

Shide committed:
[FIX] partner_second_lastname: Fix lastname2 positions

Module partner_firstname defines two lastname fields in view and this fix add lastname2 after each lastname field.

OCA-git-bot committed:
Merge PR #2211 into 18.0

Signed-off-by yajo

OCA-git-bot committed:
[BOT] post-merge updates

OCA-git-bot committed:
Merge PR #2000 into 18.0

Signed-off-by NL66278

Commit:
[UPD] Update partner_title_active.pot

OCA-git-bot committed:
[BOT] post-merge updates

Saran440 committed:
[FIX] base_location: domain invisible city in company & placeholder

OCA-git-bot committed:
Merge PR #2046 into 18.0

Signed-off-by NL66278

OCA-git-bot committed:
[BOT] post-merge updates

OCA-git-bot committed:
Merge PR #2214 into 18.0

Signed-off-by pedrobaeza

Commit:
[UPD] Update base_location.pot

OCA-git-bot committed:
[BOT] post-merge updates

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: partner-contact-18.0/partner-contact-18.0-base_location
Translate-URL: https://translation.odoo-community.org/projects/partner-contact-18-0/partner-contact-18-0-base_location/

Ryoko04 committed:
Added translation using Weblate (Japanese)

OCA-git-bot committed:
Merge PR #2064 into 18.0

Signed-off-by NL66278

Commit:
[UPD] Update partner_industry_parent.pot

OCA-git-bot committed:
[BOT] post-merge updates

mymage committed:
Added translation using Weblate (Italian)

mymage committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (17 of 17 strings)

Translation: partner-contact-18.0/partner-contact-18.0-partner_industry_secondary
Translate-URL: https://translation.odoo-community.org/projects/partner-contact-18-0/partner-contact-18-0-partner_industry_secondary/it/

mymage committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (37 of 37 strings)

Translation: partner-contact-18.0/partner-contact-18.0-base_location_nuts
Translate-URL: https://translation.odoo-community.org/projects/partner-contact-18-0/partner-contact-18-0-base_location_nuts/it/

mymage committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (5 of 5 strings)

Translation: partner-contact-18.0/partner-contact-18.0-partner_industry_parent
Translate-URL: https://translation.odoo-community.org/projects/partner-contact-18-0/partner-contact-18-0-partner_industry_parent/it/

legalsylvain committed:
[REF] partern_firstname: adopt an addon

Ryoko04 committed:
Translated using Weblate (Japanese)

Currently translated at 100.0% (6 of 6 strings)

Translation: partner-contact-18.0/partner-contact-18.0-partner_contact_address_default
Translate-URL: https://translation.odoo-community.org/projects/partner-contact-18-0/partner-contact-18-0-partner_contact_address_default/ja/

OCA-git-bot committed:
Merge PR #2117 into 18.0

Signed-off-by NL66278

Commit:
[UPD] Update partner_firstname.pot

OCA-git-bot committed:
[BOT] post-merge updates

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: partner-contact-18.0/partner-contact-18.0-partner_firstname
Translate-URL: https://translation.odoo-community.org/projects/partner-contact-18-0/partner-contact-18-0-partner_firstname/

mymage committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (25 of 25 strings)

Translation: partner-contact-18.0/partner-contact-18.0-partner_firstname
Translate-URL: https://translation.odoo-community.org/projects/partner-contact-18-0/partner-contact-18-0-partner_firstname/it/

OCA-git-bot committed:
Merge PR #2210 into 18.0

Signed-off-by NL66278

Commit:
[UPD] Update partner_contact_address_default.pot

OCA-git-bot committed:
[BOT] post-merge updates

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: partner-contact-18.0/partner-contact-18.0-partner_contact_address_default
Translate-URL: https://translation.odoo-community.org/projects/partner-contact-18-0/partner-contact-18-0-partner_contact_address_default/

OCA-git-bot committed:
Merge PR #1975 into 18.0

Signed-off-by NL66278

Commit:
[UPD] Update partner_tier_validation.pot

OCA-git-bot committed:
[BOT] post-merge updates

mymage committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (13 of 13 strings)

Translation: partner-contact-18.0/partner-contact-18.0-partner_contact_address_default
Translate-URL: https://translation.odoo-community.org/projects/partner-contact-18-0/partner-contact-18-0-partner_contact_address_default/it/

Commit:
Translated using Weblate (Spanish)

Currently translated at 100.0% (10 of 10 strings)

Translation: partner-contact-18.0/partner-contact-18.0-partner_affiliate
Translate-URL: https://translation.odoo-community.org/projects/partner-contact-18-0/partner-contact-18-0-partner_affiliate/es/

Commit:
Translated using Weblate (French)

Currently translated at 100.0% (10 of 10 strings)

Translation: partner-contact-18.0/partner-contact-18.0-partner_affiliate
Translate-URL: https://translation.odoo-community.org/projects/partner-contact-18-0/partner-contact-18-0-partner_affiliate/fr/

bosd committed:
Added translation using Weblate (Dutch)

bosd committed:
Translated using Weblate (Dutch)

Currently translated at 100.0% (3 of 3 strings)

Translation: partner-contact-18.0/partner-contact-18.0-partner_contact_type_end_user
Translate-URL: https://translation.odoo-community.org/projects/partner-contact-18-0/partner-contact-18-0-partner_contact_type_end_user/nl/

OCA-git-bot committed:
Merge PR #2084 into 18.0

Signed-off-by JordiBForgeFlow

Commit:
[UPD] Update partner_supplier_ref_sequence.pot

OCA-git-bot committed:
[BOT] post-merge updates

uvid-ana committed:
Added translation using Weblate (Croatian)

Ryoko04 committed:
Translated using Weblate (Japanese)

Currently translated at 46.1% (6 of 13 strings)

Translation: partner-contact-18.0/partner-contact-18.0-partner_contact_address_default
Translate-URL: https://translation.odoo-community.org/projects/partner-contact-18-0/partner-contact-18-0-partner_contact_address_default/ja/

mymage committed:
Added translation using Weblate (Italian)

mymage committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (4 of 4 strings)

Translation: partner-contact-18.0/partner-contact-18.0-partner_supplier_ref_sequence
Translate-URL: https://translation.odoo-community.org/projects/partner-contact-18-0/partner-contact-18-0-partner_supplier_ref_sequence/it/

ivantodorovich committed:
[IMP] partner_rank_single: less invasive constraint on rank increase

Instead of running the constraint check on `_increase_rank`, we now avoid
increasing it if the partner is already ranked in the opposite field.

This fixes an error that occurs when installing demo data of core modules:

```python-traceback
2025-12-11 11:44:05,845 35 ERROR testdb odoo.addons.account.demo.account_demo: Error while posting demo data
Traceback (most recent call last):
  File ".../odoo/addons/account/demo/account_demo.py", line 90, in _post_load_demo_data
    move.action_post()
  File ".../odoo/addons/account/models/account_move.py", line 5426, in action_post
    self._post(soft=False)
  File ".../odoo/addons/account/models/account_move.py", line 5154, in _post
    (partner | partner.commercial_partner_id)._increase_rank('supplier_rank', count)
  File ".../partner-contact/partner_rank_single/models/res_partner.py", line 23, in _increase_rank
    "supplier_rank": "customer_rank",
    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File ".../partner-contact/partner_rank_single/models/res_partner.py", line 15, in _constrains_single_rank
    raise ValidationError(
odoo.exceptions.ValidationError: A contact cannot be both a customer and a supplier.
```

OCA-git-bot committed:
Merge PR #2237 into 18.0

Signed-off-by ivantodorovich

OCA-git-bot committed:
[BOT] post-merge updates

OCA-git-bot committed:
Merge PR #2208 into 18.0

Signed-off-by pedrobaeza

OCA-git-bot committed:
[BOT] post-merge updates

yostashiro committed:
[ADD] partner_bank_acc_holder_name

OCA-git-bot committed:
Merge PR #2242 into 18.0

Signed-off-by etobella

Commit:
[UPD] Update partner_bank_acc_holder_name.pot

OCA-git-bot committed:
[BOT] post-merge updates

pere-aquarian committed:
Translated using Weblate (Catalan)

Currently translated at 100.0% (6 of 6 strings)

Translation: partner-contact-18.0/partner-contact-18.0-partner_manual_rank
Translate-URL: https://translation.odoo-community.org/projects/partner-contact-18-0/partner-contact-18-0-partner_manual_rank/ca/

OCA-git-bot committed:
Merge PR #2057 into 18.0

Signed-off-by NL66278

Commit:
[UPD] Update base_partner_sequence.pot

OCA-git-bot committed:
[BOT] post-merge updates

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: partner-contact-18.0/partner-contact-18.0-base_partner_sequence
Translate-URL: https://translation.odoo-community.org/projects/partner-contact-18-0/partner-contact-18-0-base_partner_sequence/

mymage committed:
Added translation using Weblate (Italian)

mymage committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (3 of 3 strings)

Translation: partner-contact-18.0/partner-contact-18.0-base_partner_sequence
Translate-URL: https://translation.odoo-community.org/projects/partner-contact-18-0/partner-contact-18-0-base_partner_sequence/it/

mymage committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (4 of 4 strings)

Translation: partner-contact-18.0/partner-contact-18.0-partner_vat_unique
Translate-URL: https://translation.odoo-community.org/projects/partner-contact-18-0/partner-contact-18-0-partner_vat_unique/it/

OCA-git-bot committed:
Merge PR #2086 into 18.0

Signed-off-by pedrobaeza

Commit:
[UPD] Update partner_country_lang.pot

OCA-git-bot committed:
[BOT] post-merge updates

OCA-git-bot committed:
Merge PR #2024 into 18.0

Signed-off-by sebastienbeau

Commit:
[UPD] Update partner_phone_secondary.pot

OCA-git-bot committed:
[BOT] post-merge updates

OCA-git-bot committed:
Merge PR #1905 into 18.0

Signed-off-by NL66278

Commit:
[UPD] Update partner_type_base.pot

OCA-git-bot committed:
[BOT] post-merge updates

legalsylvain committed:
[FIX] partner_firstname: Revert part of the changes of #1905.

Rational:
- #1905 introduces a migration script to force the installation of a new module named 'partner_type_base'. The migration script doesn't work, so all production instances can not update partner_firstname for the time being.
-> removing the migration script and the dependency.
- #1905 makes the computed function _compute_firstname_lastname_required in 'firstname.mixin' depending is_individual field, but this field is not present in this mixin.
-> add is_individual at firstname.mixin level.
- #1905 consider that invoicing and delivery partner can not be individual people. That assertion can make sense, but it is changing the default odoo base module. That is not the objective of partner_firstname to introduce that change.
-> the new is_individual field introduced is now more 'agnostic'. For the time being, in `res.partner`, partner.is_individual = not partner.is_company. Note that this computation can be easely overloaded.
- Revert a change in the computation of 'firstname_required' and 'lastname_required' fields, that was incorrect. A and B or C is not the same as A and (B or C).
- Remove a default value for 'partner_names_required_fields' defined in res.config.settings, as it is useless. The default value is set in the data created.

legalsylvain committed:
[IMP] partner_firstname: replace partner by more generic word 'item', to avoid confusion

OCA-git-bot committed:
Merge PR #2263 into 18.0

Signed-off-by legalsylvain

Commit:
[UPD] Update partner_firstname.pot

Commit:
[UPD] Update partner_type_base.pot

OCA-git-bot committed:
[BOT] post-merge updates

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: partner-contact-18.0/partner-contact-18.0-partner_firstname
Translate-URL: https://translation.odoo-community.org/projects/partner-contact-18-0/partner-contact-18-0-partner_firstname/

mymage committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (27 of 27 strings)

Translation: partner-contact-18.0/partner-contact-18.0-partner_firstname
Translate-URL: https://translation.odoo-community.org/projects/partner-contact-18-0/partner-contact-18-0-partner_firstname/it/

mymage committed:
Added translation using Weblate (Italian)

legalsylvain committed:
[REV] Finish to revert #1905.
Begin of the revert has been done in #2263. But after comments in #2263, it looks like it's better to full revert the initial PR. (Also the other module partner_type_base, will be removed in another PR #2266).

OCA-git-bot committed:
Merge PR #2267 into 18.0

Signed-off-by legalsylvain

Commit:
[UPD] Update partner_firstname.pot

Commit:
[UPD] Update partner_type_base.pot

OCA-git-bot committed:
[BOT] post-merge updates

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: partner-contact-18.0/partner-contact-18.0-partner_type_base
Translate-URL: https://translation.odoo-community.org/projects/partner-contact-18-0/partner-contact-18-0-partner_type_base/

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: partner-contact-18.0/partner-contact-18.0-partner_firstname
Translate-URL: https://translation.odoo-community.org/projects/partner-contact-18-0/partner-contact-18-0-partner_firstname/

mymage committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (9 of 9 strings)

Translation: partner-contact-18.0/partner-contact-18.0-partner_type_base
Translate-URL: https://translation.odoo-community.org/projects/partner-contact-18-0/partner-contact-18-0-partner_type_base/it/

bo-eska committed:
Translated using Weblate (Turkish)

Currently translated at 100.0% (3 of 3 strings)

Translation: partner-contact-18.0/partner-contact-18.0-base_partner_sequence
Translate-URL: https://translation.odoo-community.org/projects/partner-contact-18-0/partner-contact-18-0-base_partner_sequence/tr/

sysadminmatmoz committed:
Translated using Weblate (Slovenian)

Currently translated at 23.5% (4 of 17 strings)

Translation: partner-contact-18.0/partner-contact-18.0-partner_industry_secondary
Translate-URL: https://translation.odoo-community.org/projects/partner-contact-18-0/partner-contact-18-0-partner_industry_secondary/sl/

chetan137 committed:
[FIX] partner_firstname: use correct config parameter key

OCA-git-bot committed:
Merge PR #2272 into 18.0

Signed-off-by legalsylvain

OCA-git-bot committed:
[BOT] post-merge updates

OCA-git-bot committed:
Merge PR #1990 into 18.0

Signed-off-by NL66278

Commit:
[UPD] Update partner_identification_eori.pot

OCA-git-bot committed:
[BOT] post-merge updates

bo-eska committed:
Added translation using Weblate (Turkish)

bo-eska committed:
Translated using Weblate (Turkish)

Currently translated at 100.0% (3 of 3 strings)

Translation: partner-contact-18.0/partner-contact-18.0-base_country_state_translatable
Translate-URL: https://translation.odoo-community.org/projects/partner-contact-18-0/partner-contact-18-0-base_country_state_translatable/tr/

osasix committed:
[18.0][FIX] base_location_nuts: fix dc terms prefix typo

sysadminmatmoz committed:
Translated using Weblate (Slovenian)

Currently translated at 50.0% (12 of 24 strings)

Translation: partner-contact-18.0/partner-contact-18.0-base_location_geonames_import
Translate-URL: https://translation.odoo-community.org/projects/partner-contact-18-0/partner-contact-18-0-base_location_geonames_import/sl/

sysadminmatmoz committed:
Translated using Weblate (Slovenian)

Currently translated at 49.3% (39 of 79 strings)

Translation: partner-contact-18.0/partner-contact-18.0-partner_identification
Translate-URL: https://translation.odoo-community.org/projects/partner-contact-18-0/partner-contact-18-0-partner_identification/sl/

sysadminmatmoz committed:
Translated using Weblate (Slovenian)

Currently translated at 100.0% (2 of 2 strings)

Translation: partner-contact-18.0/partner-contact-18.0-partner_contact_lang
Translate-URL: https://translation.odoo-community.org/projects/partner-contact-18-0/partner-contact-18-0-partner_contact_lang/sl/

sysadminmatmoz committed:
Translated using Weblate (Slovenian)

Currently translated at 100.0% (2 of 2 strings)

Translation: partner-contact-18.0/partner-contact-18.0-partner_contact_nationality
Translate-URL: https://translation.odoo-community.org/projects/partner-contact-18-0/partner-contact-18-0-partner_contact_nationality/sl/

sysadminmatmoz committed:
Translated using Weblate (Slovenian)

Currently translated at 100.0% (3 of 3 strings)

Translation: partner-contact-18.0/partner-contact-18.0-partner_contact_birthdate
Translate-URL: https://translation.odoo-community.org/projects/partner-contact-18-0/partner-contact-18-0-partner_contact_birthdate/sl/

sysadminmatmoz committed:
Translated using Weblate (Slovenian)

Currently translated at 90.6% (87 of 96 strings)

Translation: partner-contact-18.0/partner-contact-18.0-partner_multi_relation
Translate-URL: https://translation.odoo-community.org/projects/partner-contact-18-0/partner-contact-18-0-partner_multi_relation/sl/

sysadminmatmoz committed:
Translated using Weblate (Slovenian)

Currently translated at 100.0% (3 of 3 strings)

Translation: partner-contact-18.0/partner-contact-18.0-partner_fax
Translate-URL: https://translation.odoo-community.org/projects/partner-contact-18-0/partner-contact-18-0-partner_fax/sl/

sysadminmatmoz committed:
Translated using Weblate (Slovenian)

Currently translated at 100.0% (6 of 6 strings)

Translation: partner-contact-18.0/partner-contact-18.0-partner_deduplicate_acl
Translate-URL: https://translation.odoo-community.org/projects/partner-contact-18-0/partner-contact-18-0-partner_deduplicate_acl/sl/

sysadminmatmoz committed:
Translated using Weblate (Slovenian)

Currently translated at 100.0% (2 of 2 strings)

Translation: partner-contact-18.0/partner-contact-18.0-partner_deduplicate_by_website
Translate-URL: https://translation.odoo-community.org/projects/partner-contact-18-0/partner-contact-18-0-partner_deduplicate_by_website/sl/

sysadminmatmoz committed:
Translated using Weblate (Slovenian)

Currently translated at 100.0% (4 of 4 strings)

Translation: partner-contact-18.0/partner-contact-18.0-partner_deduplicate_filter
Translate-URL: https://translation.odoo-community.org/projects/partner-contact-18-0/partner-contact-18-0-partner_deduplicate_filter/sl/

sysadminmatmoz committed:
Translated using Weblate (Slovenian)

Currently translated at 100.0% (15 of 15 strings)

Translation: partner-contact-18.0/partner-contact-18.0-partner_employee_quantity
Translate-URL: https://translation.odoo-community.org/projects/partner-contact-18-0/partner-contact-18-0-partner_employee_quantity/sl/

sysadminmatmoz committed:
Translated using Weblate (Slovenian)

Currently translated at 100.0% (34 of 34 strings)

Translation: partner-contact-18.0/partner-contact-18.0-partner_external_map
Translate-URL: https://translation.odoo-community.org/projects/partner-contact-18-0/partner-contact-18-0-partner_external_map/sl/

sysadminmatmoz committed:
Translated using Weblate (Slovenian)

Currently translated at 100.0% (14 of 14 strings)

Translation: partner-contact-18.0/partner-contact-18.0-partner_contact_department
Translate-URL: https://translation.odoo-community.org/projects/partner-contact-18-0/partner-contact-18-0-partner_contact_department/sl/

sysadminmatmoz committed:
Translated using Weblate (Slovenian)

Currently translated at 100.0% (5 of 5 strings)

Translation: partner-contact-18.0/partner-contact-18.0-partner_contact_gender
Translate-URL: https://translation.odoo-community.org/projects/partner-contact-18-0/partner-contact-18-0-partner_contact_gender/sl/

sysadminmatmoz committed:
Translated using Weblate (Slovenian)

Currently translated at 100.0% (12 of 12 strings)

Translation: partner-contact-18.0/partner-contact-18.0-partner_contact_job_position
Translate-URL: https://translation.odoo-community.org/projects/partner-contact-18-0/partner-contact-18-0-partner_contact_job_position/sl/

sysadminmatmoz committed:
Translated using Weblate (Slovenian)

Currently translated at 100.0% (24 of 24 strings)

Translation: partner-contact-18.0/partner-contact-18.0-partner_capital
Translate-URL: https://translation.odoo-community.org/projects/partner-contact-18-0/partner-contact-18-0-partner_capital/sl/

sysadminmatmoz committed:
Translated using Weblate (Slovenian)

Currently translated at 100.0% (3 of 3 strings)

Translation: partner-contact-18.0/partner-contact-18.0-base_country_state_translatable
Translate-URL: https://translation.odoo-community.org/projects/partner-contact-18-0/partner-contact-18-0-base_country_state_translatable/sl/

sysadminmatmoz committed:
Translated using Weblate (Slovenian)

Currently translated at 100.0% (3 of 3 strings)

Translation: partner-contact-18.0/partner-contact-18.0-base_partner_sequence
Translate-URL: https://translation.odoo-community.org/projects/partner-contact-18-0/partner-contact-18-0-base_partner_sequence/sl/

sysadminmatmoz committed:
Translated using Weblate (Slovenian)

Currently translated at 100.0% (17 of 17 strings)

Translation: partner-contact-18.0/partner-contact-18.0-partner_industry_secondary
Translate-URL: https://translation.odoo-community.org/projects/partner-contact-18-0/partner-contact-18-0-partner_industry_secondary/sl/

KristianSolkaer committed:
[FIX] partner_firstname: stabilize name inverse to avoid constraint errors

OCA-git-bot committed:
Merge PR #2280 into 18.0

Signed-off-by legalsylvain

OCA-git-bot committed:
[BOT] post-merge updates

mymage committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (17 of 17 strings)

Translation: partner-contact-18.0/partner-contact-18.0-partner_tier_validation
Translate-URL: https://translation.odoo-community.org/projects/partner-contact-18-0/partner-contact-18-0-partner_tier_validation/it/

OCA-git-bot committed:
Merge PR #2276 into 18.0

Signed-off-by NL66278

OCA-git-bot committed:
[BOT] post-merge updates

OCA-git-bot committed:
Merge PR #1944 into 18.0

Signed-off-by NL66278

Commit:
[UPD] Update partner_address_split.pot

OCA-git-bot committed:
[BOT] post-merge updates

victoralmau committed:
[IMP] partner_identification: Change where the mock has to be loaded

With the changes introduced in odoo/odoo@de056cc#diff-b37c7fd5520c97a29ddc59495779e7be61a66e15e85603928e27b3affb9dc31f, an error was occurring because the change validation on the model was being performed before tearDownClass was executed. Therefore, it has been modified so that the mock changes are applied in each test and cleaned up after each test. This way, the error will no longer appear and the test will run normally.

victoralmau committed:
[IMP] partner_category_type: Change where the mock has to be loaded

Related to https://github.com/odoo/odoo/commit/de056cc784a3bbe2575fd3c9e81ca62e73c362d4

victoralmau committed:
[FIX] partner_contact_age_range: Fix test

OCA-git-bot committed:
Merge PR #2291 into 18.0

Signed-off-by pedrobaeza

Commit:
[UPD] Update partner_email_check.pot

OCA-git-bot committed:
[BOT] post-merge updates

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: partner-contact-18.0/partner-contact-18.0-partner_email_check
Translate-URL: https://translation.odoo-community.org/projects/partner-contact-18-0/partner-contact-18-0-partner_email_check/

mymage committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (19 of 19 strings)

Translation: partner-contact-18.0/partner-contact-18.0-partner_email_check
Translate-URL: https://translation.odoo-community.org/projects/partner-contact-18-0/partner-contact-18-0-partner_email_check/it/
